### PR TITLE
feat: add main oak posthog client

### PR DESCRIPTION
**Pull request by mantagen**
## Description

- adapts analyticsprovider to be able to handle multiple posthog instances
- ensure the main Oak instance only gets sent Avo track events
- ensure ai-beta instance gets used for feature flags
- rename server-side client so that it's clear which PH instance it's connected to (this was when i thought thered be two serverside ... but we actually only need that one, happy to change back name if people prefer)

## Issue(s)

Fixes https://www.notion.so/oaknationalacademy/Consistent-PostHog-ID-across-labs-owa-48f99fc7818744af91c2c2dd6630a7c1?pvs=4

## How to test

1. Go to https://oak-ai-lesson-assistant-nm1c29abt.vercel-preview.thenational.academy
2. accept analytics cookies
3. Navigate around, log in etc
4. You should see events coming through to both 
  - https://eu.posthog.com/project/124 MAIN OAK 
  - https://eu.posthog.com/project/8837 AI BETA
5. if you've already been on OWA staging/previews before with cookies accepted then you should have your old MAIN OAK posthog person have new events associated with it

### CODE REVIEW

File: apps/nextjs/src/components/ContextProviders/AnalyticsProvider.tsx [hunk 1]
```diff
@@ -80,18 +77,30 @@ export type AnalyticsProviderProps = {
 };
 
 if (
+  // Main Oak posthog key
+  !process.env.NEXT_PUBLIC_POSTHOG_OAK_API_KEY ||
+  // Ai-beta posthog key
```
Comment (tomwisecodes) [inline apps/nextjs/src/components/ContextProviders/AnalyticsProvider.tsx:82]: have you added to doppler prd and stg?

**Review by tomwisecodes — APPROVED** [review]
Looks good to me

**Comment by tomwisecodes** [comment]
got a linting error though

File: apps/nextjs/src/components/ContextProviders/AnalyticsProvider.tsx [hunk 2]
```diff
@@ -55,11 +52,11 @@ export type AnalyticsContext = {
   identify: IdentifyFn;
   page: PageFn;
   reset: ResetFn;
-  posthogDistinctId: PosthogDistinctId | null;
```
Comment (codeincontext) [inline apps/nextjs/src/components/ContextProviders/AnalyticsProvider.tsx:58]: Do we no longer need this?

**Review by codeincontext — APPROVED** [review]

File: apps/nextjs/src/components/ContextProviders/AnalyticsProvider.tsx (hunk 2 above)
Comment (mantagen) [reply apps/nextjs/src/components/ContextProviders/AnalyticsProvider.tsx:58]: We don't!

Comment (mantagen) [reply apps/nextjs/src/components/ContextProviders/AnalyticsProvider.tsx:58]: I think maybe we never did

Comment (codeincontext) [reply apps/nextjs/src/components/ContextProviders/AnalyticsProvider.tsx:58]: Ok 👍

**Pull request merged**